### PR TITLE
E2E clients cleanup

### PR DESCRIPTION
**Pull request by m1kola**
### Which issue this PR addresses:

Part of the [WI 13667437](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/13667437).

### What this PR does / why we need it:

* Removes unused clients
* Switches some tests to the newer API version

### Test plan for issue:

Run E2E

### Is there any documentation that needs to be updated for this PR?

No, just test refactoring.

### CODE REVIEW

File: test/e2e/disk_encryption.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ var _ = Describe("Encryption at host should be enabled", func() {
 		ctx := context.Background()
 
 		By("getting the resource group where the VM instances live in")
-		oc, err := clients.OpenshiftClustersv20200430.Get(ctx, vnetResourceGroup, clusterName)
+		oc, err := clients.OpenshiftClustersv20220904.Get(ctx, vnetResourceGroup, clusterName)
```
Comment (m1kola) [inline test/e2e/disk_encryption.go:23]: I'll be adding additional checks to these two tests: want to check that cluster resource has appropriate fields set

**Review by cblecker — APPROVED** [review]

**Comment by bennerv** [comment]
@m1kola should we /hold this PR until we perform an ARM manifest rollout with the new API version?  

Otherwise if e2e runs against production, or we have to cut a new release with this PR it will fail since the API is not released and the client GETs using the newest API (which isn't yet released) will not work.

**Review by bennerv — DISMISSED** [review]

**Comment by m1kola** [comment]
@bennerv that makes sense, but it will block my next PR which relies on some new fields. Let's switch to 2022-04-01 for now and then we can switch again to the newer one later.

I updated this PR.

**Review by bennerv — APPROVED** [review]
lgtm

**Pull request merged**